### PR TITLE
Control video tiles carousel

### PR DESCRIPTION
**Pull request by samthor**
Adds `<web-event-carousel>` to control whether the YouTube elements are clickable and have a thumbnail.

Shows enabled if there's a valid videoId and if the day is complete.

### CODE REVIEW

File: src/site/content/en/live/index.njk [hunk 1]
```diff
@@ -97,26 +97,8 @@ date: 2020-06-31
...
-          </div>
-          <div class="w-event-carousel__day">
-            <div class="w-event-carousel__thumbnail">
-            </div>
-            <div>Day 3 — Coming soon</div>
-          </div>
-        </div>
       {% endif %}
-      {% EventTable event.days, collections.authors %}
+      {% EventTable event.days, collections.authors, event.isDuringEvent or event.isPostEvent %}
```
Comment (robdodson) [inline src/site/content/en/live/index.njk:101]: I'd suggest making the carousel its own component so the EventTable doesn't have to mess with rendering it as well and we can throw it into a conditional block above the table.

File: src/site/_includes/components/EventTable.js [hunk 2]
```diff
@@ -112,8 +113,24 @@ module.exports = (days, authorsCollection) => {
     `;
   };
 
+  const renderCarouselDay = ({title, videoId}, index) => {
+    const previewImage = videoId ? html`<img src="https://img.youtube.com/vi/${videoId}/maxresdefault.jpg"` : '';
```
Comment (robdodson) [inline src/site/_includes/components/EventTable.js:117]: I can't recall the thumbnail size but i think we can request smaller versions.

default.jpg <- 120 x 90
mqdefault.jpg <- 320 x 180

**Comment by samthor** [comment]
This is ready now.

File: src/styles/pages/_live.scss [hunk 3]
```diff
@@ -285,6 +286,13 @@
   &:focus {
     outline: 0;
     opacity: .9;
+    text-decoration: none;
+  }
+
+  &:focus {
+    .w-event-carousel__description {
+      outline: 1px solid $FOCUS_COLOR;
```
Comment (robdodson) [inline src/styles/pages/_live.scss:294]:
```suggestion
      box-shadow: inset 0 0 0 1px $FOCUS_COLOR;
```

I think this helps with z clipping on the left edge of day 1

File: src/styles/pages/_live.scss [hunk 4]
```diff
@@ -253,6 +253,7 @@
 .w-event-carousel {
   display: flex;
   overflow-x: scroll;
+  padding-bottom: 12px;
```
Comment (robdodson) [inline src/styles/pages/_live.scss:256]: on smaller screens this pushes the scroll bar closer to the tabs and away from the carousel, was that the intention? Or were you trying to create more separation between the carousel itself and the tabs?

File: src/lib/components/EventSchedule/index.js [hunk 5]
```diff
@@ -177,12 +177,22 @@ class EventSchedule extends HTMLElement {
...
     this._activeEventDay = activeEventDay;
 
+    eventDays.forEach(({isComplete, videoId}, index) => {
+      const el = this.querySelector(`[data-index="${index}"]`);
+      if (!el) {
+        return;
+      }
+      // TODO(samthor): Update description ("Coming soon" vs "All sessions")
+      if (isComplete) {
+        el.setAttribute('href', 'https://youtu.be/' + videoId);
```
Comment (robdodson) [inline src/lib/components/EventSchedule/index.js:190]: this is setting an href on the `.web-tabs__panel`, were you planning to use it for something?

File: src/lib/components/EventCarousel/index.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+        >
+          <div class="w-event-carousel__thumbnail">
+            <img
+              alt="${title}"
+              src="https://img.youtube.com/vi/${videoId}/maxresdefault.jpg"
+              width="178"
+              height="110"
+            />
+          </div>
+          <div class="w-event-carousel__description">${title}</div>
```
Comment (robdodson) [inline src/lib/components/EventCarousel/index.js:66]:
```suggestion
          <div class="w-event-carousel__description">${title} &mdash; All sessions</div>
```

File: src/site/_data/event.js [hunk 7]
```diff
@@ -404,8 +404,7 @@ const communityEvents = {
 };
 
 module.exports = {
-  isPreEvent: true,
-  isDuringEvent: false,
+  isDuringEvent: true,
```
[the remark this answers is not in the document]
Comment (robdodson) [reply src/site/_data/event.js:407]: also need to add `isPreEvent` back

File: src/lib/components/EventSchedule/index.js (hunk 5 above)
Comment (samthor) [reply src/lib/components/EventSchedule/index.js:190]: This whole section was actually my 1st attempt at this code. It didn't break so I didn't remove it. Gone now.

File: src/site/_data/event.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (samthor) [reply src/site/_data/event.js:407]: so we don't actually _use_ `isPreEvent` anywhere. Would you be fine with removing it for now?

File: src/styles/pages/_live.scss (hunk 4 above)
Comment (samthor) [reply src/styles/pages/_live.scss:256]: I've changed this to margin, I want to create some spacing between carousel and tabs. So the scroll bar now is close to the carousel.

**Pull request merged**